python/langsmith/evaluation/_arunner.py

```diff
@@ -642,6 +651,61 @@ async def astart(self) -> _AsyncExperimentManager:
             upload_results=self._upload_results,
         )
 
+    async def awith_predictions_and_evaluators(
```

Comment:
We could probably do something similar to what we do in the sync version to avoid having to duplicate logic here (basically share a semaphor)